tock/tock/middleware.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.contrib import auth
+from datetime import datetime, timedelta
+
+
+class AutoLogout():
+    def process_request(self, request):
+        # Check if user is logged in at all
+        if not request.user.is_authenticated():
+            return
+
+        fmt = '%Y%m%d%H%M%S'
+        # Compare the time of the last activity with the logout delay
+        try:
+            session_time = datetime.strptime(
+                request.session['tock_last_activity'],
+                fmt
+            )
+            if datetime.now() - session_time > \
+               timedelta(0, settings.AUTO_LOGOUT_DELAY_MINUTES * 60, 0):
```

Comment:
You could maybe make this more readable by changing the import to 
```
from django.conf.settings import AUTO_LOGOUT_DELAY_MINUTES as LOGOUT_TIME
```
or, alternately, doing the math earlier:
```
LOGOUT_TIME = settings.AUTO_LOGOUT_DELAY_MINUTES * 60
```

Which would in turn let you get this cleanly on one line:
```
if datetime.now - session_time >  timedelta(0, LOGOUT_TIME):
```
Also, I don't think the trailing 0 arg is needed is it?